db/src/tx_observer.rs

```diff
@@ -104,6 +99,7 @@ pub struct AsyncBatchExecutor {
 
 impl Command for AsyncBatchExecutor {
     fn execute(&self) {
+        // need to clone to move to a new thread.
```

Comment:
I don't believe you do; I think you need to take ownership of the boxed commands. You can do that with

```
let command_queue = ::std::mem::replace(&mut self.commands, Default::default());
```

Reply:
Sadly not that I can figure out. Because what we have here are a bunch of Trait Objects, they have no constant size and so `::std::mem::replace` doesn't accept them as arguments. I was hoping this was going to be the key - the clone here is the only reason why we need `Clone` on `Command`, and it's `Clone` that prevents me from using references to TxReports inside a `Vec`. Because we need to store them inside `AsyncBatchExecutor` and that requires defining a lifetime for the reference, and because the implementation of `Clone` on `CommandClone` requires a `'static` lifetime, the two lifetimes don't play well together. Hence storing a `Vec<Arc<Command>>`.

Reply:
https://github.com/mozilla/mentat/commit/349d3d399025d1842fce9338ba3ae1a0e1d07b4a